cmd/main.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net/url"
+	"os"
+
+	"github.com/JoaoBraveCoding/prom-storage-analysis/pkg/parser"
+	"github.com/JoaoBraveCoding/prom-storage-analysis/pkg/prom"
+)
+
+var (
+	server *string
+)
+
+func init() {
+	server = flag.String("url", "http://localhost:9090/", "server")
+}
+
+func main() {
+
+	flag.Parse()
+
+	parsedURL, err := url.Parse(*server)
+	if err != nil {
+		fmt.Println(fmt.Errorf("error while parsing server variable, %s", err))
+		os.Exit(1)
+	}
+
+	expressions := prom.GetUsedExprInRules(parsedURL)
+
+	mapOfMetrics := make(map[string]bool)
```

Comment:
(nit) a map of empty struct is the idiomatic way in Go to implement a set.

```suggestion
	mapOfMetrics := make(map[string]struct{})
```